### ISSUE TITLE
track_features in .condarc causes exception

### ISSUE THREAD

**Issue by gmnash** [conda/conda#2524]
Using the track_features config in the .condarc in conda 4.0.x causes an exception to be thrown.  The problem does not occur in conda 3.19.3.

```
rm -rf $HOME/miniconda
rm -rf Miniconda-latest-Linux-x86_64.sh

wget https://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh
./Miniconda-latest-Linux-x86_64.sh -b -p $HOME/miniconda

# name the tracked feature anything
cat > $HOME/miniconda/.condarc << EOF
channels:
   - defaults

track_features: abc
EOF

# install any package
$HOME/miniconda/bin/conda install cython -y
```

Results in the following stacktrace:

```
Traceback (most recent call last):
  File "/home/gnash/miniconda/bin/conda", line 6, in <module>
    sys.exit(main())
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/cli/main.py", line 139, in main
    args_func(args, p)
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/cli/main.py", line 146, in args_func
    args.func(args, p)
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/cli/main_install.py", line 62, in execute
    install.install(args, parser, 'install')
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/cli/install.py", line 334, in install
    update_deps=args.update_deps)
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/plan.py", line 413, in install_actions
    pkgs = r.install(specs, [d + '.tar.bz2' for d in linked], update_deps=update_deps)
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/resolve.py", line 875, in install
    pkgs = self.solve(specs, len0=len0, returnall=returnall)
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/resolve.py", line 909, in solve
    dists, new_specs = self.get_dists(specs)
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/resolve.py", line 520, in get_dists
    if not full_prune(specs, removes, optional, features):
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/resolve.py", line 503, in full_prune
    nfeats.update(self.track_features(fn))
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/resolve.py", line 586, in track_features
    return set(self.index[fn].get('track_features', '').split())
KeyError: 'b@'

```

Installing any package, not just cython, causes this error.  The key it is missing seems to be a function of the feature name.

**Comment by mcg1969** [comment]
Thanks for the report. We should be able to get this fixed relatively quickly.

**Comment by mcg1969** [comment]
@ilanschnell, it looks like you changed this behavior in https://github.com/conda/conda/commit/c6c0ea1d6181da3be5b32da78857e74124d4e8eb

Do you recall why? Looks like it was done in response to #2203, but I don't think it was actually a bug. In fact, you wrote the original code here: https://github.com/conda/conda/commit/b3d523cafbdcadefde9b6b672d67595b8d2ba5d3

@gmnash, you can fix this by writing your `track_features` entry as a JSON list:
```
track_features:
  - abc
```
I propose we support both syntaxes.

**Comment by ilanschnell** [comment]
Hmm, this was a long time ago, but since you can have multiple features I thought it would be cleaner to have them as a list.  However since the old "split string" behavior had been around for 2 years, it is probably best to do that.

**Comment by mcg1969** [comment]
OK, PR built. This was the issue I was referring to. Note the use of the trailing `@` to tell the solver about an package-free track_feature.

**Comment by gmnash** [comment]
It seems like that doesn't fix the problem in conda 4.0.5:

```
rm -rf $HOME/miniconda
rm -rf Miniconda-latest-Linux-x86_64.sh

wget https://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh
chmod +x Miniconda-latest-Linux-x86_64.sh
./Miniconda-latest-Linux-x86_64.sh -b -p $HOME/miniconda

# name the tracked feature anything
cat > $HOME/miniconda/.condarc << EOF
channels:
   - defaults

track_features: 
   - abc
EOF

# install any package
$HOME/miniconda/bin/conda install cython -y
```

The key it can't find changed slightly though:

```
Traceback (most recent call last):
  File "/home/gnash/miniconda/bin/conda", line 6, in <module>
    sys.exit(main())
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/cli/main.py", line 139, in main
    args_func(args, p)
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/cli/main.py", line 146, in args_func
    args.func(args, p)
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/cli/main_install.py", line 62, in execute
    install.install(args, parser, 'install')
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/cli/install.py", line 334, in install
    update_deps=args.update_deps)
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/plan.py", line 413, in install_actions
    pkgs = r.install(specs, [d + '.tar.bz2' for d in linked], update_deps=update_deps)
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/resolve.py", line 875, in install
    pkgs = self.solve(specs, len0=len0, returnall=returnall)
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/resolve.py", line 909, in solve
    dists, new_specs = self.get_dists(specs)
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/resolve.py", line 520, in get_dists
    if not full_prune(specs, removes, optional, features):
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/resolve.py", line 503, in full_prune
    nfeats.update(self.track_features(fn))
  File "/home/gnash/miniconda/lib/python2.7/site-packages/conda/resolve.py", line 586, in track_features
    return set(self.index[fn].get('track_features', '').split())
KeyError: 'abc@'
```

**Comment by mcg1969** [comment]
I'll verify that, but it does fix it in master, which will become 4.1 very soon.

**Comment by mcg1969** [comment]
Ah, I see. The failure happens if you choose a `track_feature` that is not represented anywhere in the package index. I'll modify the PR to fix that too.

**Comment by gmnash** [comment]
Ok.  I will try it again when 4.1 is released.  Thanks.

**Comment by mcg1969** [comment]
Here's the change to get that second problem solved: https://github.com/conda/conda/pull/2541/commits/1b53b3be30741b1e7b6365fcf0f10149b468ab94

Feel free to manually add it into your current `conda` if you're daring! :-) Thanks for the report.

**Comment by mcg1969** [comment]
I'm confident we've fixed this in 4.1, so I'll go ahead and close. But do feel free to re-open if that proves not to be the case.

**Comment by gmnash** [comment]
Tried to leave a comment before and didn't post, so apologies if this double posts.  It appears that several versions of 4.1.x have been released but are not available here: https://repo.continuum.io/miniconda/  When they be made available?
